tests/unit/unit-util.cpp

```diff
@@ -47,6 +47,28 @@ class UnitUtilTests : public ::testing::Test,
   TermVec symbols;
 };
 
+class UnitUtilDimacsTests
+    : public ::testing::Test,
+      public ::testing::WithParamInterface<SolverConfiguration>
+{
+ protected:
+  void SetUp() override
+  {
+    SolverConfiguration se = GetParam();
+    SolverConfiguration sc(se.solver_enum, true);
```

Comment:
Here you can explain that we are only using logging solvers for this test because we don't want the original formula to change -- otherwise it might no longer be a CNF.

Reply:
I am unable to reply to the comment above this one, so I am replying here. My function is taking the last element of the vector during "and elimination", recursively removing and pushing it to the end of a new vector, due to which the order of clauses change, a similar thing occurs during "or elimination". The set of clauses, in the end, are the same, so I don't think it would matter much.